app/src/controllers/accessRequest.ts

```diff
@@ -15,106 +15,95 @@ const controller = {
+          res.status(422).json({ message: 'Must provide a group to grant' });
         }
       }
 
       const isGroupUpdate = existingUser && accessRequest.grant;
       let response;
 
       if (isGroupUpdate) {
-        // Remove all user groups
-        for (const g of userGroups) {
-          let initiative = req.currentContext?.initiative as Initiative;
-          if (g.groupName === GroupName.DEVELOPER) {
-            initiative = Initiative.PCNS;
-          }
-
-          response = await yarsService.removeGroup(userResponse?.sub as string, initiative, g.groupName);
+        // Remove all user groups for initiative
+        const groupsToRemove = accessUserGroups.filter((x) => x.initiativeId === requestedGroup?.initiativeId);
+        for (const g of groupsToRemove) {
+          response = await yarsService.removeGroup(userResponse?.sub as string, g.groupId);
         }
 
         // Assign new group
-        await yarsService.assignGroup(
-          req.currentContext.bearerToken,
-          user.sub,
-          req.currentContext?.initiative as Initiative,
-          accessRequest.group as GroupName
-        );
+        await yarsService.assignGroup(req.currentContext.bearerToken, user.sub, accessRequest.groupId);
 
         // Mock an access request for the response
         response = {
           userId: userResponse?.userId,
           grant: accessRequest.grant,
-          group: accessRequest.group,
+          groupId: accessRequest.groupId,
           status: AccessRequestStatus.APPROVED
         };
-      } else if (admin) {
+      } else if (isAdmin) {
         if (accessRequest.grant) {
-          await yarsService.assignGroup(
-            req.currentContext.bearerToken,
-            user.sub,
-            req.currentContext?.initiative as Initiative,
-            accessRequest.group as GroupName
-          );
+          await yarsService.assignGroup(req.currentContext.bearerToken, user.sub, accessRequest.groupId);
           // Mock an access request for the response
           response = {
             userId: userResponse?.userId,
             grant: accessRequest.grant,
-            group: accessRequest.group,
+            groupId: accessRequest.groupId,
             status: AccessRequestStatus.APPROVED
           };
         } else {
-          // Remove requested group if provided - otherwise remove all user groups
-          const groupsToRemove = accessRequest.group ? [{ groupName: accessRequest.group }] : userGroups;
-          for (const g of groupsToRemove) {
-            let initiative = req.currentContext?.initiative as Initiative;
-            if (g.groupName === GroupName.DEVELOPER) {
-              initiative = Initiative.PCNS;
-            }
-
-            response = await yarsService.removeGroup(userResponse?.sub as string, initiative, g.groupName);
+          // Remove requested group if provided - otherwise remove all user groups for initiative
+          const groupsToRemove = accessRequest.groupId
```

Comment:
duplicate code in `processUserAccessRequest` consider helper/refactor (optional)